src/parsers/manifest/dash/node_parsers/Period.ts

```diff
@@ -50,14 +52,79 @@ export interface IPeriodAttributes {
   xlinkActuate? : string;
 }
 
+/**
+ * Parse the EventStream node to extract Event nodes and their
+ * content.
+ * @param {Element} element
+ */
+function parseEventStream(element: Element): IStreamEvent[] {
+  const streamEvents: IStreamEvent[] = [];
+  const attributes: { schemeId?: string;
+                      timescale: number;
+                      value?: string; } =
+                      { timescale: 1 };
+
+  for (let i = 0; i < element.attributes.length; i++) {
+    const attribute = element.attributes[i];
+    switch (attribute.name) {
+      case "schemeIdUri":
+        attributes.schemeId = attribute.value;
+        break;
+      case "timescale":
+        attributes.timescale = parseInt(attribute.value, 10);
+        break;
+      case "value":
+        attributes.value = attribute.value;
+        break;
+      default:
+        break;
+    }
+  }
+
+  for (let k = 0; k < element.childNodes.length; k++) {
+    const node = element.childNodes[k];
+    if (node.nodeName === "Event" &&
+        node.nodeType === Node.ELEMENT_NODE) {
+      let presentationTime;
+      let duration;
+      let id;
+      const eventAttributes = (node as Element).attributes;
+      for (let j = 0; j < eventAttributes.length; j++) {
+        const attribute = eventAttributes[j];
+        switch (attribute.name) {
+          case "presentationTime":
+            const pts = parseInt(attribute.value, 10);
+            presentationTime = pts / attributes.timescale;
```

Comment:
This type of logic is generally done elsewhere in the parser. `node_parsers` where much "dumber" as their task was mostly to convert the XML into a JS Object.

Reply:
Yes !